src/components/AuthStoreSetter.js

```diff
@@ -0,0 +1,28 @@
+import _ from 'lodash/fp'
+import { useEffect } from 'react'
+import { useAuth } from 'react-oidc-context'
+import { processUser, reloadAuthToken } from 'src/libs/auth'
+import { useOnMount } from 'src/libs/react-utils'
+import { authStore } from 'src/libs/state'
+
+
+const AuthStoreSetter = () => {
+  const auth = useAuth()
+
+  useOnMount(() => authStore.update(_.set(['authContext'], auth)))
+
+  useEffect(() => {
+    const cleanupFns = [
+      auth.events.addAccessTokenExpiring(reloadAuthToken),
+      auth.events.addUserLoaded(processUser),
+      auth.events.addUserUnloaded(processUser)
+    ]
+    return _.over(cleanupFns)
```

Comment:
As written, this will not return a function to invoke upon cleanup, but the results of running all of the `cleanupFns`. I believe the cleanup will be invoked when as soon as this is run, not when `auth` changes.

Side note: TIL about `_.over`

```suggestion
    return () => _.over(cleanupFns)
```

Reply:
@petesantos are you sure? I think `_.over([...])` itself returns a function, and doesn't invoke anything immediately. I did a quick test:
```
var f1 = () => console.log("f1")
var f2 = () => console.log("f2")

// prints nothing
var x = _.over([f1, f2])

// prints f1, f2
x()

// prints nothing
var y = () => _.over([f1, f2])

// prints nothing
y()

// prints f1, f2
y()()
```

Reply:
(And I think all the `cleanupFns` are of type `() => void`)

Reply:
You are right! I haven't used `over` before and when I looked at the doc I misread it